cclib/parser/turbomoleparser.py

```diff
@@ -109,19 +112,60 @@ def extract(self, inputfile, line):
             nmo = int(line.split('=')[1])
             self.set_attribute('nbasis', nmo)
             self.set_attribute('nmo', nmo)
-
+            
+        # The DFT functional.
+        # This information is printed by dscf but not in an easily parsible format, so we'll take it from the control file instead...
+        # Additionally, turbomole stores functional names in lower case. This looks odd, so we'll convert to uppercase (?)

```

Comment:
This is a fine first approximation, then you discover wB97 and related...but I think that's an edge case not worth worrying about now.